giftless/auth/github.py

```diff
@@ -238,67 +280,226 @@ def from_dict(cls, data: Mapping[str, Any]) -> "Config":
 
 
 # CORE AUTH
-@dataclasses.dataclass(frozen=True, slots=True)
```

Comment:
This whole bit is an interesting and elegant way to approach user caching.  I like it.